src/infra/build-agents/README.md

```diff
@@ -1,9 +1,81 @@
 # Self-hosted agent pools with Azure DevOps
 
+- [Overview](#overview)
+- [Infrastructure components](#infrastructure-components)
+- [Connect to Jump Server via Bastion](#connect-to-jump-server-via-bastion)
+
+---
+
+## Overview
+
 This part of the repository contains the required Bicep template to deploy the infrastructure used for self-hosted Azure DevOps build agent pools, based on Virtual Machine Scale Sets (VMSS).
 
 Furthermore, it deploys a VMSS which is used as Jump Servers to connect to the private resources, such as private AKS clusters, for debugging etc. To connect to the Jump Servers, Azure Bastion is used. This way, even the Jump Servers do not require any public IPs.
 
+To deploy the infrastructure for the self-hosted Agents and all supporting services such as Jump Servers and private DNS zones, a ready-to-use Bicep template plus the corresponding ADO Pipeline is included in this repository. Bicep is being used for this part of the infrastructure instead of Terraform, because using Terraform could create a "chicken-and-egg" problem with the state storage account which requires public access. All further resources are then created using Terraform.
+
+## Infrastructure components
+
+The Bicep template ([`main.bicep`](./main.bicep)) deploys the following infrastructure components:
```

Comment:
I'm missing here some documentation on "how to use" the template - az command and description of the required parameters.

Reply:
Should I just link to the pipeline? The pipeline is there for the deployment

Reply:
how about now?